test/e2e/es/mutation_test.go

```diff
@@ -227,6 +232,57 @@ func TestMutationWithLargerMaxUnavailable(t *testing.T) {
 	RunESMutation(t, b, mutated)
 }
 
+func TestMutationWhileLoadTesting(t *testing.T) {
+	b := elasticsearch.NewBuilder("test-while-load-testing").
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources).
+		WithHTTPLoadBalancer()
+
+	// force a rolling upgrade through label change
+	var pt corev1.PodTemplateSpec
+	pt.Labels = map[string]string{"some_label_name": "some_new_value"}
+	mutated := b.WithNoESTopology().
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources).
+		WithPodTemplate(pt)
+
+	// keep hitting ES endpoints at high rate during pod cycling to catch any downtime
+	attacker := vegeta.NewAttacker(vegeta.KeepAlive(false))
+	var metrics vegeta.Metrics
+	w := test.NewOnceWatcher(
+		"load test",
+		func(k *test.K8sClient, t *testing.T) {
+			var ip string
+			for {
+				svc, err := k.GetService(b.Elasticsearch.Namespace, esv1.HTTPService(b.Elasticsearch.Name))
```

Comment:
this seems easy enough to understand but a small `// wait for the LoadBalancer IP to be available` wouldn't hurt the reader :)